xds/internal/server/listener_wrapper.go

```diff
@@ -213,7 +217,7 @@ type listenerWrapper struct {
 	rdsHandler *rdsHandler
 	// rdsUpdates are the RDS resources received from the management
 	// server, keyed on the RouteName of the RDS resource.
-	rdsUpdates map[string]xdsclient.RouteConfigUpdate // TODO: if this will be read in accept, this will need a read lock as well.
+	rdsUpdates unsafe.Pointer
```

Comment:
Maybe include the actual underlying type in a comment, like we do here: https://github.com/grpc/grpc-go/blob/master/internal/internal.go#L66 and what is the rationale behind using an `unsafe.Pointer`.

Reply:
Added. This used to be a map[string]xdsclient.RouteConfigUpdate, protected by a mutex due to reading it in Accept() and writing to it on any rds update. However, Doug mentioned that this is unoptimal because it is slower than using an unsafe.Pointer and also because accepted connections will have to wait on the rds update write to finish, which will cause this flow to have many waits.